City-of-Helsinki/drupal-hdbt
# Uhf 7462 drupal bundle

# [UHF-7462](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7462)

Testing to eslint-airbnb-base and eslint-prettier default configs (more or less).


## What was done

Added
```
 "eslint-config-airbnb-base": "15.0.0",
  "eslint-config-prettier": "^8.5.0",
  "eslint-import-resolver-webpack": "0.13.2",
  "eslint-plugin-import": "^2.25.3",
 "eslint-plugin-prettier": "^4.2.1",
```

Should now lint and format according to eslint-airbnb-base. A few overrides have been added to avoid extensive but trivial stylistic code changes.


**Prototype for using eslint-airbnb-base from drupal core as a default linting setup for hel.fi JS code.**
This pull request serves as a discussion topic for further development. 
This includes adding **react** and **typescript** support.

Added

Changing lint rules naturally caused a bunch of errors in all js files. 
These have been fixed in the commit. ~95% of the changes were either automatic or "disable this rule for  now because we need to talk about it", such as anonymous functions in IIFE lines and generator/iterator- syntax. Manual work consists of naming anonymous functions as per lint rules.

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-7462-drupal-bundle`

- run `npm install`


## How to test

- open a javascript file. Make random changes, do some silly and maybe buggy code lines and save.
- run `npm run lint:js` . It should give you list of lint warnings and errors
- run  `npm `run format:js` It should fix all formatting issues if you made any.

- If your editor is configured to use eslint, you should see lint errors while you work (if you configured it so). 

For VSCode, you may need to set the path to eslint. It is recommended you use the project's local version to ensure there are not version mismatches.

File->Preferences->Settings (Workspace settings.json) -> 
```
  {
  ...
   "eslint.nodePath": "node_modules/eslint",
   "editor.formatOnSave": true
  }
```

if you set formatOnSave, your editor should now format your code according to  eslint/prettier rules.

## Designers review


* [X ] This PR does not need designers review

## Other PRs


* Link to other PR
